### PR TITLE
show-cert-revokation option added

### PR DESCRIPTION
**Pull request by z-eos**
show-cert-revokation option allows to see cert revocation state

> EXAMPLE:
> ./easyrsa show-cert-revokation CERT-TO-TEST
> 
> 
> Note: using Easy-RSA configuration from: ./vars
> C = US, ST = CA, L = SanFrancisco, O = Fort-Funston, OU = Cache Segment, CN = CERT-TO-TEST, emailAddress = security@foo.bar
> error 23 at 0 depth lookup: certificate revoked
> error /path/to/easyrsa3/pki/issued/CERT-TO-TEST.crt: verification failed

### CODE REVIEW

File: easyrsa3/easyrsa [hunk 1]
```diff
@@ -1739,7 +1758,10 @@ case "$cmd" in
 		show req "$@"
 		;;
 	show-cert)
-		show cert "$@"
+	    show cert "$@"
+	    ;;
+	show-cert-revokation)
+		show cert-revokation "$@"
```
Comment (ecrist) [inline easyrsa3/easyrsa]: The ca+crl.pem file should be cleaned up/removed.

**Comment by z-eos** [comment]
On Sun, Jan 12, 2020 at 1:36 AM Eric Crist <notifications@github.com> wrote:

> *@ecrist* requested changes on this pull request.
> ------------------------------
>
> In easyrsa3/easyrsa
> <https://github.com/OpenVPN/easy-rsa/pull/309#discussion_r365546276>:
>
> > @@ -1,4 +1,5 @@
>  #!/bin/sh
> +# -*- mode: sh; mode: follow -*-
>
> This is a python convention, isn't it?
>
not, it is emacs convention
just the same thing as it is for vim here
https://github.com/z-eos/easy-rsa/blob/master/easyrsa3/easyrsa#L1778


> ------------------------------
>
> In easyrsa3/easyrsa
> <https://github.com/OpenVPN/easy-rsa/pull/309#discussion_r365546394>:
>
> > @@ -1185,6 +1188,12 @@ CRL Generation failed.
>  An updated CRL has been created.
>  CRL file: $out_file
>  "
> +	cat $EASYRSA_PKI/ca.crt $EASYRSA_PKI/crl.pem > $EASYRSA_PKI/ca+crl.pem
> +	notice "\
> +An updated CA+CRL has been created.
> +CA+CRL file: $EASYRSA_PKI/ca+crl.pem
>
> Frowning on the + character in the file name.
>
definitely it can be anything else, but it is used for FreeRADIUS in this
format


> ------------------------------
>
> In easyrsa3/easyrsa
> <https://github.com/OpenVPN/easy-rsa/pull/309#discussion_r365547797>:
>
> > @@ -1739,7 +1758,10 @@ case "$cmd" in
>  		show req "$@"
>  		;;
>  	show-cert)
> -		show cert "$@"
> +	    show cert "$@"
> +	    ;;
> +	show-cert-revokation)
> +		show cert-revokation "$@"
>
> The ca+crl.pem file should be cleaned up/removed.
>
it doesn't as well as crl itself
of course this can be done latter by hands ... as well as all actions ...
but it is not convenient :)


-- 
xmpp://zeus@gnu.org.ua
GPG: 0FF7A32A

File: easyrsa3/easyrsa [hunk 2]
```diff
@@ -1415,6 +1424,17 @@ Run easyrsa without commands for usage help."
 		verify_ca_init
 		in_file="$EASYRSA_PKI/issued/${name}.crt"
 		format="x509"
+ 	elif [ "$type" = "cert-revokation" ]; then
+	    local crl="$EASYRSA_PKI/ca+crl.pem"
```
Comment (TinCanTech) [inline easyrsa3/easyrsa]: Easyrsa does not allow use of `local`

**Comment by z-eos** [comment]
new request has been issued

**Pull request closed without merging**